python/paddle/v2/framework/evaluator.py

```diff
@@ -0,0 +1,59 @@
+import paddle.v2.framework.op as op
+import numpy as np
+import paddle.v2.framework.core as core
+
+
+def avg_accumulate(accumulated_var, per_eval, num_batches, place):
+    t = np.array(accumulated_var.get_tensor())
+    t[0] += per_eval[0]
+    accumulated_var.get_tensor().set([t[0] / float(num_batches)], place)
+
+
+class Evaluator(object):
+    def __init__(self,
+                 scope,
+                 operator='accuracy',
+                 input='Inference',
+                 label='Label',
+                 output='Output',
+                 place=core.CPUPlace()):
+        """
+        create an evaluator for evaluating the inference.
+        NOTE: default run on CPUPlace(), running on GPUPlace doesn't improve performance much.
+
+        :param scope: the scope instance contains the input.
+        :type scope: paddle.v2.framework.core.scope
+        :param operator: operator name for caculating the evaluation for each mini-batch.
+        :type operator: string
+        :param input: output variable name of forward network.
+        :type input: string
+        :param label: variable name of label
+        :type label: string
+        """
+        self.scope = scope
+        self.place = place
+        self.output_name = output
+        self.num_batches = 0
+        # create variable to store accumulated evaluator output
+        eval_name = ''.join([operator, "@Eval"])
+        if scope.find_var(eval_name):
+            raise Exception("evaluator already exist in scope: %s" % eval_name)
+        self.accumulated_var = scope.var(eval_name)
+        t = self.accumulated_var.get_tensor()
+        t.set_dims((1, ))
+        t.set([0.0], place)
+        # self.accumulated_var = block.create_var(block, name=eval_name, shape=(1,))
+        # self.accumulated_var.get_tensor().set([0.0])
+        # create operator of evaluation
+        var_map = dict()  # var name -> variable
+        var_map[input] = [input]
+        var_map[label] = [label]
+        var_map[output] = [output]
+        self.op = op.Operator(operator, **var_map)
```

Comment:
Currently, we can not create operator immediately out of Python `Block` and `Program`.
https://github.com/PaddlePaddle/Paddle/blob/1f11f773bf761171288b165984bc26a379fe1db8/python/paddle/v2/framework/framework.py#L430-L446  

May I just merge this work into codebase and let me polish it to fit in the refactorization code? : )
Our book porting work need this feature eagerly. @typhoonzero @qingqing01

Reply:
Sure, Thanks. Please also refer to some of my thoughts: https://github.com/PaddlePaddle/Paddle/issues/5157